projects/ad2s1210_iio/app/ad2s1210_iio.c

```diff
@@ -0,0 +1,619 @@
+/***************************************************************************//**
+ *   @file    ad2s1210_iio.c
+ *   @brief   Implementation of AD2S1210 IIO application interfaces
+********************************************************************************
+ * Copyright (c) 2023 Analog Devices, Inc.
+ * Copyright (c) 2023 BayLibre, SAS.
+ * All rights reserved.
+ *
+ * This software is proprietary to Analog Devices, Inc. and its licensors.
+ * By using this software you agree to the terms of the associated
+ * Analog Devices Software License Agreement.
+*******************************************************************************/
+
+/******************************************************************************/
+/***************************** Include Files **********************************/
+/******************************************************************************/
+
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+#include <errno.h>
+
+#include "iio.h"
+#include "iio_types.h"
+#include "iio_trigger.h"
+
+#include "ad2s1210_iio.h"
+#include "app_config.h"
+#include "ad2s1210_user_config.h"
+#include "common.h"
+#include "no_os_error.h"
+#include "no_os_util.h"
+
+/******** Forward declaration of getter/setter functions ********/
+static int iio_ad2s1210_attr_get(void *device, char *buf, uint32_t len,
+				 const struct iio_ch_info *channel,
+				 intptr_t priv);
+
+static int iio_ad2s1210_attr_set(void *device, char *buf, uint32_t len,
+				 const struct iio_ch_info *channel,
+				 intptr_t priv);
+
+/******************************************************************************/
+/********************* Macros and Constants Definition ************************/
+/******************************************************************************/
+#define AD2S1210_CHN_ATTR(_name, _priv) {\
+	.name = _name,\
+	.priv = _priv,\
+	.show = iio_ad2s1210_attr_get,\
+	.store = iio_ad2s1210_attr_set\
+}
+
+#define AD2S1210_CH(_name, _idx, _type, _ch_out) {\
+	.name = _name, \
+	.ch_type = _type,\
+	.ch_out = _ch_out,\
+	.indexed = true,\
+	.channel = 0,\
```

Comment:
Should this be '_idx' ?

Reply:
Originally i had it as _idx, but if we want to be in par with the linux kernel driver attributes it should be 0. 

These are the linux kernel attributes, if you check the suffix of the channel is always 0. and i guess it makes some sense as its the position and velocity of one resolver device

			angl0:  (input, index: 0, format: be:U16/16>>0)
			5 channel-specific attributes found:
				attr  0: hysteresis value: 1
				attr  1: hysteresis_available value: 0 1
				attr  2: label value: position
				attr  3: raw value: 12579
				attr  4: scale value: 0.000095874
			anglvel0:  (input, index: 1, format: be:S16/16>>0)
			3 channel-specific attributes found:
				attr  0: label value: velocity
				attr  1: raw value: 9
				attr  2: scale value: 0.023968449
			timestamp:  (input, index: 2, format: le:S64/64>>0)
			phase0:  (input)
			1 channel-specific attributes found:
				attr  0: label value: synthetic reference
			altvoltage0:  (output)
			3 channel-specific attributes found:
				attr  0: frequency value: 10000
				attr  1: frequency_available value: [2000 250 20000]
				attr  2: label value: excitation